### DISCUSSION TITLE
Does the `#[watch]` macro replace `#[track = ...]` for simple use cases?

### DISCUSSION THREAD

**Discussion by chriskilding** [Relm4/Relm4#519] [Q&A] [answered]
I'm trying to understand why there are two macros in Relm4 which, superficially at least, seem to do the same thing: `#[watch]` and `#[track = <condition>]`.

Is there any reason why we would use `#[track = "true"]` instead of `#[watch]`?

Or is `#[watch]` the way to go for all simple use cases, and we should only use `track` for complex conditions?

**Comment by AaronErhardt** [comment, marked as answer]
> Is there any reason why we would use #[track = "true"] instead of #[watch]?

`#[track = "true"]` is indeed identical to `#[watch]`. `#[watch]` is the simpler variant for small components and `#[track = "..."]` should be used for more complex components with several properties that are updated to reduce the amount of view updates to a minimum.

**Reply by chriskilding** [reply]
Great - in that case might it also be worth updating the book to clarify this? (And possibly the examples too - it sounds like they should all use `#[watch]` unless they need to track custom conditions.)

**Reply by AaronErhardt** [reply]
Yes, updating the book would be a good idea, although I don't exactly remember when those attributes are introduced. AFAIK the examples do not use track except for macro_reference which is explicitly there to show all macro features.
